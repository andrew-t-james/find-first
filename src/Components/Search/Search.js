import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor() {
    super();
    this.state = {
      query: ''
    };
  }

  handleUpdate = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  render() {
    const { query } = this.state;

    return (
      <div className="search">
        <input
          type="text"
          value={query}
          name="query"
          onChange={this.handleUpdate}
          className="search__input"
          placeholder="Search Jobs"
        />
      </div>
    );
  }
}

export default Search;
