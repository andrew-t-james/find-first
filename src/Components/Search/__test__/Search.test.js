import React from 'react';
import { shallow } from 'enzyme';
import { Search } from '../Search';

describe('<Search />', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Search />));

  describe('handleUpdate', () => {
    test('should update state onChange', () => {
      const mockState = {
        query: ''
      };
      const mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'query',
          value: 'my hot job'
        }
      };

      wrapper.instance().handleUpdate(mockEvent);
      expect(wrapper.state('query')).toBe('my hot job');
    });
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
