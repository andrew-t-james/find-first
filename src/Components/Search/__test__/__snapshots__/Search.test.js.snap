// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="search__input"
        name="query"
        onChange={[Function]}
        placeholder="Search Jobs"
        type="text"
        value=""
      />,
      "className": "search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "search__input",
        "name": "query",
        "onChange": [Function],
        "placeholder": "Search Jobs",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search__input"
          name="query"
          onChange={[Function]}
          placeholder="Search Jobs"
          type="text"
          value=""
        />,
        "className": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search__input",
          "name": "query",
          "onChange": [Function],
          "placeholder": "Search Jobs",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
